contracts/proto0/assets/CTokenAssetP0.sol

```diff
@@ -1,43 +1,74 @@
 // SPDX-License-Identifier: BlueOak-1.0.0
-pragma solidity 0.8.4;
+pragma solidity 0.8.9;
 
-import "../interfaces/IMain.sol";
-import "./AssetP0.sol";
+import "contracts/proto0/interfaces/IMain.sol";
+import "contracts/proto0/assets/AssetP0.sol";
+import "contracts/libraries/Fixed.sol";
+import "contracts/proto0/libraries/Oracle.sol";
+
+// cToken initial exchange rate is 0.02
 
 // https://github.com/compound-finance/compound-protocol/blob/master/contracts/CToken.sol
 interface ICToken {
+    /// @dev From Compound Docs: The current exchange rate, scaled by 10^(18 - 8 + Underlying Token Decimals).
     function exchangeRateCurrent() external returns (uint256);
 
+    /// @dev From Compound Docs: The current exchange rate, scaled by 10^(18 - 8 + Underlying Token Decimals).
     function exchangeRateStored() external view returns (uint256);
 
     function underlying() external view returns (address);
 }
 
 contract CTokenAssetP0 is AssetP0 {
+    using FixLib for Fix;
+
     // All cTokens have 8 decimals, but their underlying may have 18 or 6 or something else.
     constructor(address erc20_) AssetP0(erc20_) {}
 
-    function updateRedemptionRate() external virtual override {
+    function updateRates() external virtual override {
         ICToken(_erc20).exchangeRateCurrent();
     }
 
-    function redemptionRate() public view override returns (uint256) {
-        return ICToken(_erc20).exchangeRateStored() * 10**(18 - fiatcoinDecimals());
+    /// @return {qFiatTok/qTok}
+    function rateFiatcoin() public view override returns (Fix) {
+        Fix rate = _exchangeRateRelativeToGenesis(); // {fiatTok/tok}
+        // {qFiatTok/qTok} = {fiatTok/tok} * {qFiatTok/fiatTok} / {qTok/tok}
+
+        int8 shiftLeft = int8(fiatcoinDecimals()) - int8(decimals());
+        return rate.mul(toFix(1, shiftLeft));
```

Comment:
Hrm. Do we want `Fix.shift(shiftLeft)`? Should be both faster and clearer than this.

Reply:
Yes...we do

Reply:
solidity won't let us take advantage of the `using` trick in quite the way we want it to. Options are `toFixShift` or `FixLib.shift`. I prefer the latter from the perspective of using the library, but from the library's perspective, it's pretty out of place to make it internal.